src/ts/router/router.ts

```diff
@@ -0,0 +1,29 @@
+import { PageList, PAGE } from './pages';
+import * as model from '../model';
+
+const route = function (): void {
+  let [, path] = location.hash.split('#');
+  if (!path) path = PAGE.products.path;
+
+  const currentView = PageList.find(page => path === page.path);
+  document.querySelectorAll(`a[data-link]`)?.forEach(el => el.classList.remove('active'));
+  document.querySelector(`a[data-link="${path}"]`)?.classList.add('active');
```

Comment:
`data-*`없이도 이런 로직을 구현하는데 큰 문제는 없지 않을까요?
또한 옵셔널 체이닐 보다는 `if`문 전체로 감싸던가 `is` 타입 가드로 고려해보실 수 있겠어요!

Reply:
```suggestion
  const isAnchorElement = (anchorElement: any): anchorElement is HTMLAnchorElement => {
    return (anchorElement as HTMLAnchorElement) != undefined;
  };

  if (isAnchorElement(document.querySelector('a.active'))) {
    document.querySelector('a.active')!.classList.remove('active');
  }
  if (isAnchorElement(document.querySelector(`a[href="#${path}"]`))) {
    document.querySelector(`a[href="#${path}"]`)!.classList.add('active');
  }
```

제가 옵셔널 체이닝을 했던 이유는 element를 select 해오면 `Object is possibly 'null'.ts(2531)` 이 에러가 계속 떠서 null check하기 위함이었는데요!

코멘트 받고 해본 아래 방식으로 시도를 해봤는데요.
1. 위의 방식으로 함수 추출하여 null check
 (이게 is 타입가드 방식이 맞는지 확신은 안서지만 공식문서 참고해서..)
2. 위의 방식으로 함수 추출하여 객체 내부에 classList property가 있는지 확인
3. 함수 추출하지 않고, if 분기로 element null check

이렇게 해도 에러가 안사라져서요...  우선 Non-null 연산자(`!`)를 사용했어요!
사실 이 연산자를 사용하면 분기를 굳이 걸 필요가 없는 것 같아서 분기문 제거도 고민해보고 있어요. 이 연산자 사용이 지양되는게 아니라면 이 편이 중복도 없고, 간결해보여서요 👀

장황하게 썼지만 질문 드리고 싶은건 두 가지인데요!
- [[Using type predicates]](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) 말씀하신 타입가드가 이게 맞을까요?
- 혹시 `Non-null`연산자는 사용이 지양되나요? 
  제가 구현한 부분에 한해서는 무조건 있는 element라서 문제가 없어보여서요!

Reply:
- Using type predicates 말씀하신 타입가드가 이게 맞을까요?
  - 넵 맞습니다!
- 혹시 Non-null연산자는 사용이 지양되나요?
  - 지양하는게 맞습니다. TS에게 사용자의 타입을 장제하는 것이니까요
  - 다만 DOM, Event 타입 등 런타임에 영향이 있어 결정에 어려움이 있어서 어쩔 수 없이 사용되기도 합니다.
  - 그리하여 타입 가드를 사용해야하는거죠~!

Reply:
아하, 감사합니다ㅎㅎㅎ 🙏 
제가 타입가드를 제대로 활용을 못해서 에러가 사라지지 않고 있었던 것 같네요!
핸드북을 뜨문뜨문 읽어서 방법을 못찾은것 같은데 우선은 더 읽어보고 다시 시도해보겠습니다~!